### DISCUSSION TITLE
dynamic array of objects

### DISCUSSION THREAD

**Discussion by Csepregi** [react-hook-form/react-hook-form#3167] [Q&A]
Hi all, 
https://codesandbox.io/s/stupefied-engelbart-p6fbn?file=/src/App.js

I could not figure out how can I add , remove, dynamically to state array object as I select from dropdown.
the object look like this:
[
{
"streamkKind": "HLS",
"profileUrl": "https://amazonnews.hls"
}
]
Can be null, one, the previous example, or two like that:
[
{
"streamkKind": "HLS",
"profileUrl": "https://amazonnews.hls"
},
{
"streamkKind": "DASH",
"profileUrl": "https://amazonnews.dash"
}
]
any tipp, advice, recommendations are welcome

**Comment by MikeyAlmighty** [comment]
@Csepregi Hiya,
I'd recomnmend to use the controller from `react-hook-form`, which wraps around react-select. (A lot easier to work with controlled inputs).
https://react-hook-form.com/api#Controller
Then it's easier to pull the values from the formContext beit `watch` or `getValues`.
I'll post a codesandbox later today after work

**Reply by Csepregi** [reply]
@MikeyAlmighty thanks, I will check asap

**Reply by Csepregi** [reply]
I have to use it with checkbox, so I can handle the add, remove better, next to dropdowns there are checkbox, if checked, add to object the two values. Default values if nothing was selected from dropdown. 
I have problems with how to handle values with checkbox.
